lib/fastlane/actions/swiftlint.rb

```diff
@@ -1,11 +1,45 @@
 module Fastlane
   module Actions
     class SwiftlintAction < Action
+      VALID_REPORTERS = ['xcode', 'json', 'csv']
+
       def self.run(params)
         if `which swiftlint`.to_s.length == 0 and !Helper.test?
           raise "You have to install swiftlint using `brew install swiftlint`".red
         end
-        Actions.sh("swiftlint")
+
+        config_file = '.swiftlint.yml'
+        backup(config_file)
+        update_config(config_file, params)
+
+        command = 'swiftlint'
+        command << " > #{params[:output_file]}" if params[:output_file]
+        Actions.sh(command)
+
+      ensure
+        restore(config_file)
+      end
+
+      def self.backup(config_file)
+        FileUtils.cp(config_file, "#{config_file}.back", preserve: true) if File.exist? config_file
```

Comment:
First reason is below,
> SwiftLint can switch reporter by using .swiftlint.yml.
> So I make backup and replace reporter when executing swiftlint.

Second reason is,
I hope to no diff between executing `swiftlint` before / after.
If some changes occured, make some noise when hitting `git status`.
I got better idea is `backup and restore`.